packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx

```diff
@@ -219,6 +226,30 @@ const CompleteResetPasswordContainer = ({
     }
   };
 
+  const maybeNotifyOAuthClient = async (accountResetData: AccountResetData) => {
+    if (!isOAuthIntegration(integration)) {
+      return;
+    }
+
+    const { error, redirect, code, state } = await finishOAuthFlowHandler(
+      accountResetData.uid,
+      accountResetData.sessionToken,
+      accountResetData.keyFetchToken,
+      accountResetData.unwrapBKey
+    );
+
+    if (oAuthDataError || error) {
```

Comment:
If there is an `oAuthDataError`, we should not render this component because we can't successfully run `finishOAuthFlowHandler`, see in other files where we `return` based on that.

Sorry that I haven't had time to test this, I can circle back on Tuesday. One other thought I have is, are we already running `finishOAuthFlowHandler` elsewhere in the reset PW flow prior to reaching this page, like `ResetPasswordConfirmed`? If we can avoid hitting a couple of our endpoints again that's definitely ideal.

Reply:
We do run `finishOAuthFlowHandler` in `ResetPasswordConfirmed` --> that page is hit for non-2FA users with OAuth integration who are continuing on to a RP after their password reset. That's probably the right spot to move this webchannel message, as long as the message is handle elsewhere for 2FA users who need to sign back in after the password reset.

Reply:
Sorry @vpomerleau @LZoog , I didn't see your responses here earlier - thank you both for pointing me in the right direction!

Moving this into `ResetPasswordConfirmed` makes sense. When following the code paths though, I see that we would only get to `ResetPasswordConfirmed` if [we are in a non-sync path and resetting without recovery keys][0]. What I'm not able to follow is why we treat the with and without recovery keys differently for this.

I've put up another patch here which works, where I move the code to this new place and remove the `!integration.isSync()` check: https://github.com/jonalmeida/fxa/commit/e3fc4168013577b1cb462734788823c30eaca023

If this looks right to you, then I'll update this PR accordingly and also duplicate the patch for the [`ResetPasswordWithRecoveryKeyVerified`](https://github.com/mozilla/fxa/blob/8d553d93dbdff090bc290241e5a305c6d5a560d7/packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx#L112) path as well.

[0]: https://github.com/mozilla/fxa/blob/8d553d93dbdff090bc290241e5a305c6d5a560d7/packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx#L123-L128

Reply:
The reasoning for excluding Sync here is (I suppose) mainly for Sync desktop. For Sync desktop, we want to navigate to `/settings` after a successful password reset and not show the `ResetPasswordConfirmed` which is really just intended to be used when we are navigating away from accounts and on to a relying party. IIUC, on mobile the tab would just close once the password reset is completed so it's not an issue?

I just tested on nightly with oauth sync (with [these dev-launcher instructions](https://github.com/mozilla/fxa/blob/8d553d93dbdff090bc290241e5a305c6d5a560d7/packages/fxa-dev-launcher/README.md?plain=1#L15)) after removing the sync check, and there is currently no redirect. I'm not sure if Sync is intending to change that and provide a redirect for use after sign in.

![image](https://github.com/user-attachments/assets/abec47bf-18db-46e2-bb74-923e5eede727)

@LZoog suggested that we could perhaps be more specific and use `isDesktopSync()` instead of `isSync()`

Reply:
I also haven't tested your branch to see the full changes with the webChannel message, so you may have this flow already covered. Apologies if I'm missing some context!

Reply:
> I'm not sure if Sync is intending to change that and provide a redirect for use after sign in.

Sync oauth desktop should do whatever `fx_desktop_v3` is - sounds like we want that to redirect to `/settings`. I think you're on the right track Jonathan but that `if` check might need to be modified so we call `sensitiveDataClient.setData('accountResetData', sensitiveData);` if `isOAuthIntegration`, but only call `navigateWithQuery('/reset_password_verified'` if `!integration.isSync()`? (What's this flow do in mobile, does the web view just close?)

Reply:
Cool, this aligns with what I read and discovered today, so adding the if-statement works for oauth desktop.

> (What's this flow do in mobile, does the web view just close?)
It should in a typical login flow, but I tested this and seems like we navigate to settings and do not sign-in today. With this patch but now it does work. 🙂